scripts/generate_syncset.py

```diff
@@ -27,33 +55,72 @@ def get_all_yaml_obj(file_paths):
         objects = get_yaml_all(file)
         for obj in objects:
             yaml_objs.append(obj)
+    yaml_objs = sorted(yaml_objs)
     return yaml_objs
 
+def process_yamls(name, directory, obj):
+    o = copy.deepcopy(obj)
+    # Get all yaml files as array of yaml objects
+    yamls = get_all_yaml_obj(get_all_yaml_files(directory))
+    if len(yamls) == 0:
+        return
+
+    yamls = sorted(yamls)
+    for y in yamls:
+        if 'patch' in y:
+            if not 'patches' in o['spec']:
```

Comment:
We check for `patches` and `resources` subkeys, but is it safe to assume `spec` exists?

Also, we're performing this check on every iteration of the loop, but `o` hasn't changed. Could we just check `o` outside the loop and setup necessary keys first?